src/domain/entities/LandingNode.ts

```diff
@@ -92,44 +97,69 @@ export const updateLandingNodes = (
         })
         .compact()
         .value();
+
+    return updatedNodes.map(node => applyFavicon(node));
+};
+
+const applyFavicon = (parent: LandingNode): LandingNode => {
+    const spreadFaviconToChildren = (children: LandingNode[], favicon: string): LandingNode[] => {
+        return _.map(children, child => {
+            return {
+                ...child,
+                favicon: favicon,
+                children: spreadFaviconToChildren(child.children, favicon),
+            };
+        });
+    };
+
+    return {
+        ...parent,
+        children: spreadFaviconToChildren(parent.children, parent.favicon),
+    };
 };
 
-/* Return a redirect URL if there is only one visible action on primary nodes */
-export function getPrimaryRedirectUrl(
+// Return
+// a redirect URL if there is only one visible action on primary nodes
+// a redirect page id when there is only one visible action on primary nodes
+export function getPrimaryRedirectNodes(
     landingNode: LandingNode,
     options: { actions: Action[]; user: User }
-): Url | undefined {
+): { redirectUrl: Maybe<Url>; redirectPageId: Maybe<string> } {
     const { actions, user } = options;
 
     const actionsById = _.keyBy(actions, action => action.id);
     const showAllActions = false;
     const isRoot = true;
```

Comment:
**[Don't change it]** Good for describing what param is👌🏼 When the number of params increases as in this case, we use `(options: { })` when declaring the function